Makefile

```diff
@@ -93,17 +93,50 @@ operator-source:
 	done
 
 .PHONY: catalog
-catalog: operator-source
-	for DIR in $(SOURCE_DIR)/**/; do \
-		eval $$($(MAKE) -C $$DIR env --no-print-directory); \
-		./scripts/gen_operator_csv.py $$DIR $$OPERATOR_NAME $$OPERATOR_NAMESPACE $$OPERATOR_VERSION $(IMAGE_REGISTRY)/$(IMAGE_REPOSITORY)/$$OPERATOR_NAME:v$$OPERATOR_VERSION $(CHANNEL) || (echo "Failed to generate, cleaning up catalog-manifests/$$OPERATOR_NAME/$$OPERATOR_VERSION" && rm -rf catalog-manifests/$$OPERATOR_NAME/$$OPERATOR_VERSION && exit 3); \
+catalog: manifestdir operator-source
+	@unset OPERATOR_NAME OPERATOR_VERSION OPERATOR_NAMESPACE OPERATOR_IMAGE_URI ;\
+	for operatorrepo in $(OPERATORS); do \
+		operator="$$(echo $$operatorrepo | cut -d / -f2)" ;\
+		echo "Building catalog for $$operator in $(SOURCE_DIR)/$$operator" ;\
+		eval $$($(MAKE) -C $(SOURCE_DIR)/$$operator env --no-print-directory); \
+		if [[ -z "$${OPERATOR_NAME}" || -z "$${OPERATOR_NAMESPACE}" || -z "$${OPERATOR_VERSION}" ]]; then \
+			echo "Couldn't determine OPERATOR_NAME, OPERATOR_NAMESPACE or OPERATOR_VERSION from $(SOURCE_DIR)/$$operator. make env output follows" ; \
+			$(MAKE) -C $(SOURCE_DIR)/$$operator env ; \
+			unset OPERATOR_NAME OPERATOR_VERSION OPERATOR_NAMESPACE OPERATOR_IMAGE_URI ;\
+			exit 3 ;\
+		else \
+			./scripts/gen_operator_csv.py $(SOURCE_DIR)/$$operator $$OPERATOR_NAME $$OPERATOR_NAMESPACE $$OPERATOR_VERSION $(IMAGE_REGISTRY)/$(IMAGE_REPOSITORY)/$$OPERATOR_NAME:v$$OPERATOR_VERSION $(CHANNEL) 1>/dev/null ;\
+			if [[ $$? -ne 0 ]]; then \
+				echo "Failed to generate, cleaning up catalog-manifests/$$OPERATOR_NAME/$$OPERATOR_VERSION" ;\
+				rm -rf catalog-manifests/$$OPERATOR_NAME/$$OPERATOR_VERSION ;\
+				unset OPERATOR_NAME OPERATOR_VERSION OPERATOR_NAMESPACE OPERATOR_IMAGE_URI ;\
+				exit 3; \
+			fi ;\
+			unset OPERATOR_NAME OPERATOR_VERSION OPERATOR_NAMESPACE OPERATOR_IMAGE_URI ;\
```

Comment:
Lots of unset.  Suggest moving this to the beginning of each loop iteration.  It only needs done when initializing this loop.  If a cleanup is desired, one outside the loop execution is fine too.